frontend/src/citizen-frontend/income-statements/IncomeStatements.tsx

```diff
@@ -104,18 +105,30 @@ export default function IncomeStatements() {
 
   useEffect(() => loadData(), [loadData])
 
-  const onRemove = useCallback(
-    (id: UUID) =>
-      removeIncomeStatement(id).then((res) => {
-        if (res.isFailure) {
-          setErrorMessage({
-            title: t.income.errors.removeFailed,
-            type: 'error',
-            resolveLabel: t.common.ok
+  const [rowToDelete, setRowToDelete] = useState<UUID>()
+  const [deleting, setDeleting] = useState(false)
```

Comment:
Otherwise looks good, but I was stuck thinking whether this should be one `useState` instead of  two. Something like `undefined | UUID | true` for idle / confirming / deleting.